src/hotspot/share/opto/parse1.cpp

```diff
@@ -921,7 +921,11 @@ void Compile::return_values(JVMState* jvms) {
     kit.inc_sp(-ret_size);  // pop the return value(s)
     kit.sync_jvms();
     Node* res = kit.argument(0);
-    if (tf()->returns_inline_type_as_fields()) {
+    if (res->isa_InlineType() && VectorSupport::skip_value_scalarization(res->as_InlineType()->inline_klass()->get_InlineKlass())) {
```

Comment:
I think in general multi-field fields should be scalarised differently from normal field, since they represent an array and ideally should be packed together on the stack instead of passing separately. As a result, for now it may be better to reject scalarisation of types with multi-fields altogether.

Reply:
Agree.